DokuwikiXMLExport.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+/**
+ * This is the Dokuwiki export made by Dominik Brader for FINDOLOGIC.
+ * If any bugs occur, please submit a new issue
+ * @see https://github.com/findologic/dokuwiki-plugin-findologic-xml-export/issues/new
+ */
+if (!defined('DOKU_INC')) {
+    define('DOKU_INC', realpath(dirname(__FILE__) . '/../../../') . '/');
+}
+
+require_once(DOKU_INC . 'inc/init.php');
+require(__DIR__ . '/vendor/autoload.php');
+
+use FINDOLOGIC\Export\Exporter;
+use FINDOLOGIC\Export\Data\Name;
+use FINDOLOGIC\Export\Data\Summary;
+use FINDOLOGIC\Export\Data\Description;
+use FINDOLOGIC\Export\Data\Price;
+use FINDOLOGIC\Export\Data\Url;
+use FINDOLOGIC\Export\Data\Ordernumber;
+use FINDOLOGIC\Export\Data\DateAdded;
+use FINDOLOGIC\Export\Data\Attribute;
+
+class DokuwikiXMLExport
+{
+
+    /**
+     * Default value for a price. DokuWiki pages do not have a price and this is just a placeholder.
+     * FINDOLOGIC requires the price attribute, so this is the reason why it is exported.
+     */
+    const PRICE_PLACEHOLDER = '0.0';
+
+    /**
+     * This value is needed to tell FINDOLOGIC this is a category.
+     */
+    const CATEGORY_KEY = 'cat';
+
+    /**
+     * Generate the entire XML Export based on the Dokuwiki metadata.
+     *
+     * @param $start integer Determines the first item (offset) to be exported.
+     * @param $count integer Determines the interval size / number of items to be exported.
+     * @return string Returns the XML.
+     */
+    public function generateXMLExport($start, $count)
+    {
+        $indexer = new Doku_Indexer();
+        $pagesAndDeletedPages = $indexer->getPages();
+
+        $excludedPages = $this->splitConfigToArray($conf['plugin']['findologicxmlexport']['excludePages']);
+        $pages = null;
+        foreach ($pagesAndDeletedPages as $page) {
+            if (p_get_metadata($page)['description']) { // Only get pages with content
+                if (!in_array($page, $excludedPages)) { // Exclude pages from config
+                    $pages[] = $page;
+                }
+            }
+        }
+        $exporter = Exporter::create(Exporter::TYPE_XML, $count);
+
+        $total = count($pages);
+
+        // The count can't be higher then the total number of pages.
+        $count = min($total, $count);
+
+
+        // If count + start is higher then total, then something went totally wrong.
+        if (($count + $start) > $total) {
+            throw new \InvalidArgumentException("Call method " . __METHOD__ . " in class " . __CLASS__ . " failed while trying to validate start and count values");
```

Comment:
Exception messages will automatically log the method and class, so no need to print manually add that.